### PR TITLE
docker: Add ubi9 docker static dockerfile

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [x] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/3586

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# Set up jenkins user
+RUN useradd -m -d /home/jenkins jenkins
+RUN mkdir /home/jenkins/.ssh
+RUN echo "Jenkins_User_SSHKey" > /home/jenkins/.ssh/authorized_keys
+RUN chown -R jenkins /home/jenkins/.ssh
+RUN chmod -R og-rwx /home/jenkins/.ssh
+# RUN service ssh start
+CMD ["/usr/sbin/sshd","-D"]
+RUN dnf install -y git make gcc xorg-x11-server-Xvfb libXrender libXi libXtst fontconfig fakeroot procps-ng hostname diffutils
+RUN yum install -y coreutils --allowerasing curl
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9:50]: This has currently got a mix of `dnf` and `yum` - we should test with just `dnf` for consistency

**Review by sxa — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Install SSL Test packages
+RUN yum install -y gnutls gnutls-utils libnss3.so nss nss-tools
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9:52]: Remove libnss3.so as per https://github.com/adoptium/infrastructure/issues/3586#issuecomment-2162406210

```suggestion
RUN dnf install -y gnutls gnutls-utils nss nss-tools
```

**Comment by sxa** [comment]
Noting also that the UBI8 file (which this ubi9 one is based on) also has similar "features" to the ones mentioned in my review.

@steelhead31 Do you recall why `libnss3.so` was passed to `yum` in this dockerfile? (Ref: https://github.com/adoptium/infrastructure/pull/3169) It looks like that doesn't work on UBI8 either - it's also in the playbooks for RPM-based distributions, although I haven't yet found any it'll work on on aarch64.

**Comment by steelhead31** [comment]
> Noting also that the UBI8 file (which this ubi9 one is based on) also has similar "features" to the ones mentioned in my review.
> 
> @steelhead31 Do you recall why `libnss3.so` was passed to `yum` in this dockerfile? (Ref: #3169) It looks like that doesn't work on UBI8 either - it's also in the playbooks for RPM-based distributions, although I haven't yet found any it'll work on on aarch64.

Nope, I can't recall , it will have been something to do with a missing dependency or build from source of some kind, but its possibly redundant now. Id say remove it , and see if we encounter any other issues as a reminder.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9 (hunk 1 above)
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9]: Changed to dnf

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9 (hunk 2 above)
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.ubi9]: Removed

**Review by sxa — APPROVED** [review]
LGTM

**Review by steelhead31 — APPROVED** [review]
LGTM

**Pull request merged**